client/reader/controller.js

```diff
@@ -213,9 +241,34 @@ module.exports = {
 		);
 	},
 
+	feedDiscovery: function( context, next ) {
+		var FeedStore = require( 'lib/feed-store' ),
+			FeedStoreActions = require( 'lib/feed-store/actions' ),
+			FeedError = require( 'reader/feed-error' );
+
+		if ( ! context.params.feed_id.match( /^\d+$/ ) ) {
+			FeedStoreActions.discover( context.params.feed_id );
+
+			FeedStore.on( 'change', function() {
+				var feedId = FeedStore.getByUrl( context.params.feed_id );
```

Comment:
Done, now we'll redirect to the `read/feeds/:feed_id` URL instead of displaying the feed stream. 

This solves the problem with displaying duplicate content (the feed stream) on two (or more) different URLs.